src/plugins/bar-graph/legend-secondary-row.tsx

```diff
@@ -1,29 +1,71 @@
 import React from 'react';
 import classNames from 'classnames';
+import { observer } from 'mobx-react';
+import { Button, Menu, MenuButton, MenuItem, MenuList } from '@chakra-ui/react';
 import { useBarGraphModelContext } from './bar-graph-content-context';
 import { displayValue, isMissingData } from './bar-graph-utils';
+import { clueDataColorInfo } from '../../utilities/color-utils';
 
 interface IProps {
   attrValue: string;
 }
 
-export function LegendSecondaryRow({attrValue}: IProps) {
+export const LegendSecondaryRow = observer(function LegendSecondaryRow ({attrValue}: IProps) {
   const model = useBarGraphModelContext();
   if (!model) return null;
 
   const missingData = isMissingData(attrValue);
   const display = displayValue(attrValue);
+  const backgroundColor = model.secondaryAttribute
+    ? model.colorForSecondaryKey(attrValue)
+    : model.primaryAttributeColor;
+
+  const handleColorSelect = (color: string) => {
+    if (model.secondaryAttribute) {
+      model.setSecondaryAttributeKeyColor(attrValue, color);
+    } else {
+      model.setPrimaryAttributeColor(color);
+    }
+  };
 
   return (
     <div key={attrValue} className="secondary-value">
-      <div className="color-button">
-        <div className="color-swatch" style={{ backgroundColor: model.getColorForSecondaryKey(attrValue) }} />
-      </div>
+      <Menu placement="auto">
+        <MenuButton as={Button} unstyle="true" data-testid="color-menu-button">
+          <div className="color-button">
+            <div className="color-swatch" style={{ backgroundColor }} />
+          </div>
+        </MenuButton>
+        <MenuList
+          bg="white"
+          border="none"
+          borderRadius="5px"
+          boxShadow="0 0 5px 0 rgba(0, 0, 0, 0.35)"
+          className="color-menu-list"
+          data-testid="color-menu-list"
+          display="grid"
+          gap={0}
+          gridTemplateColumns="repeat(2, 1fr)"
```

Comment:
Can this be moved to CSS?

Reply:
@bgoldowsky As I understand it, it's not recommended to apply styles to Chakra UI elements via CSS. It's better to use Chakra UI's built-in theming and styling system. I think ideally we'd set up a Chakra UI theme for CLUE, like we've done for CODAP. I thought about doing that but this story was kind of shoehorned into this sprint and I don't think there's time to work that out as part of this work. I think adding these styles this way is still preferred to doing so via separate CSS rules. Do you disagree?

Reply:
Yeah, Chakra is not really fully integrated, I've noticed that before.  If using CSS is not recommended for these elements you can ignore my comment.